### PR TITLE
pythonPackages.kaldi-active-grammar: fix build

### LINKED ISSUE #203089: kaldi-active-grammar: "cannot load library"
### Describe the bug
I'm trying to run the example script https://raw.githubusercontent.com/daanzu/kaldi-active-grammar/master/examples/full_example.py, but I get the following error:

```console
Kaldi-Active-Grammar v3.1.0:
    If this free, open source engine is valuable to you, please consider donating
    https://github.com/daanzu/kaldi-active-grammar
    Disable message by calling `kaldi_active_grammar.disable_donation_message()`
Traceback (most recent call last):
  File "/home/silvio/kaldi/full_example.py", line 10, in <module>
    compiler = kaldi_active_grammar.Compiler(model_dir=model_dir, tmp_dir=tmp_dir)
  File "/nix/store/incraiihd7bdwvrgh46kz5648bf4ks27-python3-3.9.12-env/lib/python3.9/site-packages/kaldi_active_grammar/compiler.py", line 270, in __init__
    NativeWFST.init_class(
  File "/nix/store/incraiihd7bdwvrgh46kz5648bf4ks27-python3-3.9.12-env/lib/python3.9/site-packages/kaldi_active_grammar/wfst.py", line 202, in init_class
    cls.init_ffi()
  File "/nix/store/incraiihd7bdwvrgh46kz5648bf4ks27-python3-3.9.12-env/lib/python3.9/site-packages/kaldi_active_grammar/ffi.py", line 35, in init_ffi
    cls._lib = _ffi.init_once(cls._init_ffi, cls.__name__ + '._init_ffi')
  File "/nix/store/incraiihd7bdwvrgh46kz5648bf4ks27-python3-3.9.12-env/lib/python3.9/site-packages/cffi/api.py", line 749, in init_once
    result = func()
  File "/nix/store/incraiihd7bdwvrgh46kz5648bf4ks27-python3-3.9.12-env/lib/python3.9/site-packages/kaldi_active_grammar/ffi.py", line 40, in _init_ffi
    return _ffi.dlopen(_library_binary_path)
  File "/nix/store/incraiihd7bdwvrgh46kz5648bf4ks27-python3-3.9.12-env/lib/python3.9/site-packages/cffi/api.py", line 150, in dlopen
    lib, function_cache = _make_ffi_library(self, name, flags)
  File "/nix/store/incraiihd7bdwvrgh46kz5648bf4ks27-python3-3.9.12-env/lib/python3.9/site-packages/cffi/api.py", line 832, in _make_ffi_library
    backendlib = _load_backend_lib(backend, libname, flags)
  File "/nix/store/incraiihd7bdwvrgh46kz5648bf4ks27-python3-3.9.12-env/lib/python3.9/site-packages/cffi/api.py", line 827, in _load_backend_lib
    raise OSError(msg)
OSError: cannot load library '/nix/store/w6qjl5zfygzyica619imp2pvpy7hydyf-kaldi-kag-v2.1.0/lib/libkaldi-dragonfly.so': libstdc++.so.6: cannot open shared object file: No such file or directory.  Additionally, ctypes.util.find_library() did not manage to locate a library called '/nix/store/w6qjl5zfygzyica619imp2pvpy7hydyf-kaldi-kag-v2.1.0/lib/libkaldi-dragonfly.so'
```

### Steps To Reproduce
1. `nix develop` into this flake:
```nix
{
  inputs = {
    nixpkgs.url = "github:NixOS/nixpkgs/22.05";
    flake-utils.url = "github:numtide/flake-utils";
  };

  outputs = { self, nixpkgs, flake-utils }:
    flake-utils.lib.eachDefaultSystem (system:
      let
        pkgs = nixpkgs.legacyPackages.${system};

        my-python = pkgs.python3;
        python-with-my-packages = my-python.withPackages (p: with p; [
#          dragonfly
          kaldi-active-grammar
        ]);
      in {
        devShell = pkgs.mkShell {
         packages = [
           python-with-my-packages
          ];
        };
      });
}
```
3. `wget https://raw.githubusercontent.com/daanzu/kaldi-active-grammar/master/examples/full_example.py`
4. (Adapt paths in script)
5. `python full_example.py`

### Expected behavior
I'm trying out the software for the first time, so I didn't know what to expect, but I guess I didn't expect this error.

### Additional context
I tried nixpkgs master first, but there, `kaldi-kag` wouldn't build.

### Notify maintainers
@ckiee 

### Metadata

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.76, NixOS, 22.11 (Raccoon)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.11.0`
 - channels(silvio): `"home-manager"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

### PR DESCRIPTION
**Pull request by ckiee**
###### Description of changes
Fixes #203089; cc @Lykos153 @mweinelt.

Snuck `backports-zoneinfo` in as it's an indirect dependency and something I ran into because my test shell had `python38`. Can drop.
 
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

ZHF: #199919

### CODE REVIEW

File: pkgs/development/python-modules/kaldi-active-grammar/fork.nix [hunk 1]
```diff
@@ -13,15 +13,33 @@
...
       rev = "0bca6e76d24647427356dc242b0adbf3b5f1a8d9";
       sha256 = "1802rr14a03zl1wa5a0x1fa412kcvbgprgkadfj5s6s3agnn11rx";
     };
     buildInputs = [ zlib ];
-  }); in
+  });
+
+  # commit f07a0615ea82351bbddc0799364774cb350ff759 "openblas: 0.3.20 -> 0.3.21"
+  # ... breaks header compatibility with the outdated kaldi-active-grammar tree.
+  old-openblas = openblas.overrideAttrs (prev: rec {
```
Comment (risicle) [inline pkgs/development/python-modules/kaldi-active-grammar/fork.nix:31]: Weird things might happen if this package were imported alongside `numpy`

Comment (ckiee) [reply pkgs/development/python-modules/kaldi-active-grammar/fork.nix:31]: Action recommended? It's contained to just this derivation so I don't anticipate much fallout.

Comment (risicle) [reply pkgs/development/python-modules/kaldi-active-grammar/fork.nix:31]: Took a bit of digging but revealed opencv having similar issues:

```
  * Use LAPACK_xxxx to transparently (via predefined lapack macros) deal with pre and post 3.9.1 versions.
  * LAPACK 3.9.1 introduces LAPACK_FORTRAN_STRLEN_END and modifies (through preprocessing) the declarations of the following functions used in opencv
  *        sposv_, dposv_, spotrf_, dpotrf_, sgesdd_, dgesdd_, sgels_, dgels_
  * which end up with an extra parameter.
  * So we also need to preprocess the function calls in opencv coding by prefixing them with LAPACK_.
  * The good news is the preprocessing works fine whatever netlib's LAPACK version.
```

https://fossies.org/linux/opencv/cmake/OpenCVFindLAPACK.cmake

Comment (risicle) [reply pkgs/development/python-modules/kaldi-active-grammar/fork.nix:31]: Adding

```
    substituteInPlace src/matrix/cblas-wrappers.h \
      --replace stptri_ LAPACK_stptri \
      --replace dtptri_ LAPACK_dtptri \
      --replace sgetrf_ LAPACK_sgetrf \
      --replace dgetrf_ LAPACK_dgetrf \
      --replace sgetri_ LAPACK_sgetri \
      --replace dgetri_ LAPACK_dgetri \
      --replace sgesvd_ LAPACK_sgesvd \
      --replace dgesvd_ LAPACK_dgesvd \
      --replace ssptri_ LAPACK_ssptri \
      --replace dsptri_ LAPACK_dsptri \
      --replace ssptrf_ LAPACK_ssptrf \
      --replace dsptrf_ LAPACK_dsptrf
```

to `postPatch` allows it to build successfully with 0.3.21

Comment (SuperSandro2000) [reply pkgs/development/python-modules/kaldi-active-grammar/fork.nix:31]: Same with openfst. We should not merge this in the current state.

Comment (risicle) [reply pkgs/development/python-modules/kaldi-active-grammar/fork.nix:31]: The openfst one isn't new and I think it's much less likely to produce a collision - numpy is everywhere on the other hand.

File: pkgs/development/python-modules/backports-zoneinfo/default.nix [hunk 2]
```diff
@@ -69,6 +69,9 @@ buildPythonPackage rec {
     "test_folds_from_utc"
     # backports.zoneinfo._common.ZoneInfoNotFoundError: 'No time zone found with key Eurasia/Badzone'
     "test_bad_keys"
+    # (on python38) AssertionError: datetime.timedelta(days=-1, seconds=84879) != datetime.timedelta(days=-1, seconds=84900) : 1880-07-31 00:00:00-00:25:21
```
Comment (risicle) [inline pkgs/development/python-modules/backports-zoneinfo/default.nix:72]: Do these only fail on python 3.8?

File: pkgs/development/python-modules/kaldi-active-grammar/fork.nix (hunk 1 above)
Comment (ckiee) [reply pkgs/development/python-modules/kaldi-active-grammar/fork.nix]: Pushed @risicle's suggestion; @SuperSandro2000: KAG uses a openfst fork, so it's not the same.

(I tried typing this with the working build and *it* does work, but effort. Spent like two minutes figuring out I need to say "location <500ms> say <200ms> super <500ms> tab" or a longer string)

File: pkgs/development/python-modules/backports-zoneinfo/default.nix (hunk 2 above)
Comment (ckiee) [reply pkgs/development/python-modules/backports-zoneinfo/default.nix:72]: Mhm

Comment (SuperSandro2000) [reply pkgs/development/python-modules/backports-zoneinfo/default.nix:72]: We no longer support python 3.8

File: pkgs/development/python-modules/kaldi-active-grammar/fork.nix (hunk 1 above)
Comment (ckiee) [reply pkgs/development/python-modules/kaldi-active-grammar/fork.nix]: I also tried it before saying that and it does not work. The fork is based on 1.7, we are on 1.8 - I am not interested in debugging this, and it's out of the scope of my changes: cleaning [openfst] up.

File: pkgs/development/python-modules/backports-zoneinfo/default.nix (hunk 2 above)
Comment (ckiee) [reply pkgs/development/python-modules/backports-zoneinfo/default.nix:72]: `hexa` on matrix told me the same. Remove or keep? Apparently the python date packages consistently have tests like this.

Comment (risicle) [reply pkgs/development/python-modules/backports-zoneinfo/default.nix:72]: My point was just to only disable these packages conditionally for python<=3.8

Comment (ckiee) [reply pkgs/development/python-modules/backports-zoneinfo/default.nix]: Ended up dropping, ran my exact-I-hope thing from before but it doesn't reproduce. Maybe I rebased onto master.

**Review by risicle — APPROVED** [review]
Builds happily, aarch64-linux

**Pull request merged**